docs/rules/indent.md

```diff
@@ -0,0 +1,82 @@
+# Validate Indentation (indent)
+
+This option validates a specific tab width for your code in block statements.
```

Comment:
@nzakas `why indentation is used' - to make code readable?
added various styles mentioning